business_product_location/models/business_product_location.py

```diff
@@ -0,0 +1,160 @@
+                                        'Business Usage')
+
+
+class BusinessProductLocationReport(models.AbstractModel):
+    _name = 'report.business_product_location.report_bpl'
+
+    @api.multi
+    def render_html(self, data=None):
+        """ Get the list of product excepted in the location and the list of
+            product in the location with the quantity
+        """
+        docs = self.env['stock.location'].browse(self._ids)
+        product_obj = self.env['product.product']
+
+        less_product_by_loc = {}
+        more_product_by_loc = {}
+
+        for location in docs:
+            if location.business_usage_id:
+                location_name = '%s - %s' % (location.name,
+                                             location.business_usage_id.name)
+                more_product_by_loc[location_name] = {}
+                less_product_by_loc[location_name] = {}
+                for product_line in location.business_usage_id.product_ids:
+                    less_product_by_loc[location_name][
+                        product_line.product_id.id] = {
+                        'product_name': product_line.product_id.display_name,
+                        'excepted_qty': product_line.product_qty,
+                        'real_qty': 0,
+                        'outgoing_qty': 0,
+                        'strong': True}
+                # get product in this location
+                product_in_loc = product_obj.with_context(location=location.id
+                                                          ).search(
+                    [('qty_available', '>', 0)])
+                for product in product_in_loc:
+                    product_to_update = less_product_by_loc[location_name].get(
+                        product.id, {'product_name': product.display_name,
+                                     'excepted_qty': 0,
+                                     'real_qty': 0,
+                                     'outgoing_qty': 0,
+                                     'strong': True})
+                    product_to_update['real_qty'] = product.qty_available
+                    product_to_update['outgoing_qty'] = product.outgoing_qty
+                    if float_compare(
+                        product_to_update['excepted_qty'],
+                        product_to_update['real_qty'] -
+                            product_to_update['outgoing_qty'],
+                            precision_rounding=1) == 0:
+                        product_to_update['strong'] = False
+                    if product.id not in less_product_by_loc[location_name
+                                                             ].keys():
+                        more_product_by_loc[location_name][
+                            product.id] = product_to_update
+                    else:
+                        less_product_by_loc[location_name][
+                            product.id] = product_to_update
```

Comment:
Consider split your method (render_html) into different functions. The indentation makes it hard to read.